examples/test_util/EclFile.cpp

```diff
@@ -16,490 +16,835 @@
    along with OPM.  If not, see <http://www.gnu.org/licenses/>.
    */
 
+#include "EclFile.hpp"
+#include "EclUtil.hpp"
+
+#include <functional>
 #include <string>
 #include <string.h>
 #include <sstream>
 #include <iterator>
 #include <iomanip>
 #include <algorithm>
 
-#include "EclFile.hpp"
+#include <opm/common/ErrorMacros.hpp>
 
 
 // anonymous namespace for EclFile
 
 namespace {
-  
-bool isEOF(std::fstream *fileH){
-    
-    /* try to read 4 byte (integer), if passed eof of, return true.
-       if not, seek back to current file possition  */
-    
-    int num;
-    long int pos=fileH->tellg();
-    fileH->read((char *)&num, sizeof(num));
-    
-    if (fileH->eof()){
-        return true;
-    } else {
-        fileH->seekg (pos);
-        return false;
-    }
-}
-
-int flipEndianInt(const int &num) {
-
-   unsigned int tmp=__builtin_bswap32(num);
-   
-   return (int)tmp;
-}
-
-float flipEndianFloat(const float &num) {
 
-    float value=num;
 
-    char *floatToConvert = reinterpret_cast<char*>(&value);
-    std::reverse(floatToConvert, floatToConvert+4);
-    
-    return value;
-}
+bool isFormatted(const std::string& filename)
+{
+    int p = filename.find_last_of(".");
+    int l = filename.length();
 
-double flipEndianDouble(const double &num) {
-    double value=num;
-    
-    char *doubleToConvert = reinterpret_cast<char*>(&value);
-    std::reverse(doubleToConvert, doubleToConvert+8);
+    std::string rootN = filename.substr(0,p);
+    std::string extension = filename.substr(p,l-p);
 
-    return value;
+    return extension.substr(1,1) == "F";
```

Comment:
This is close - but the case of multiple formatted summary data files create a list of files: `CASE.A0000, CASE.A0001, CASE.A0002, ...`

Reply:
added "A".